backend/src/fetcher.py

```diff
@@ -369,6 +369,16 @@ def __init__(self, packit_id: int) -> None:
         self.packit_id = packit_id
         self.copr_url = f"{self.packit_api_url}/copr-builds/{self.packit_id}"
         self.koji_url = f"{self.packit_api_url}/koji-builds/{self.packit_id}"
+        self._provider = None
+
+    @property
+    def provider(self):
+        """
+        Cached so that we don't send HTTP requests for every call
+        """
+        if not self._provider:
+            self._provider = self._get_correct_provider()
+        return self._provider
```

Comment:
tip: there is `cached_property` in functools that do for you exactly this (you only need to return something, the rest handles the decorator)